backend/app/engine/projects/rewards/capped_quadratic_funding.py

```diff
@@ -94,9 +113,44 @@ def calculate_project_rewards(
 
             total_allocated += plain_allocation
 
+        return rewards, project_rewards_sum, total_allocated
+
+    def calculate_project_rewards(
+        self, payload: ProjectRewardsPayload
+    ) -> ProjectRewardsResult:
+        """
+        Calculate rewards for projects using plain quadratic funding formula with capped funding.
+        """
+        before_capped_matched_by_addr, _ = self.capped_mr.compute(
+            payload, payload.before_allocations
+        )
+        merged_allocations = payload.before_allocations + payload.user_new_allocations
+        capped_matched_by_addr, grouped_allocations = self.capped_mr.compute(
+            payload, merged_allocations
+        )
+
+        rewards, project_rewards_sum, total_allocated = self._aggregate_rewards(
+            payload, capped_matched_by_addr, grouped_allocations
+        )
+
+        project_addresses = list(
+            map(lambda x: x.project_address, payload.user_new_allocations)
+        )
+        user_new_allocations = sum(
+            list(map(lambda x: x.amount, payload.user_new_allocations))
+        )
+
+        leverage = self.leverage.calculate_individual_leverage(
+            before_capped_matched_by_addr,
+            capped_matched_by_addr,
+            project_addresses,
+            user_new_allocations,
+        )
```

Comment:
as this logic becomes more complicated, we need to create some kind of flow chart that covers that algo and all the edge cases. I think we should dicsuss the dev documentation once we are done with all the tickets and bugs for this epoch. Maybe @adam-gf can contribute to this as a part of this onboarding process. Up until it was very scattered.

Reply:
i totally agree, we need a documentation for that; me or @adam-gf can create a one when the dust settles